presto-main/src/main/java/com/facebook/presto/operator/aggregation/OptimizedTypedSet.java

```diff
@@ -47,6 +52,7 @@ public class OptimizedTypedSet
     private static final SelectedPositions EMPTY_SELECTED_POSITIONS = positionsList(new int[0], 0, 0);
 
     private final Type elementType;
+    private final Optional<MethodHandle> elementIsDistinctFrom;
```

Comment:
How difficult would it be to make it required and always use `isDistinctFrom`?

Reply:
actually, map_concat doesn't otherwise check that map keys are not indeterminate or null (caused failing tests because of this), and it's a bit challenging to get it to do that because it's a varargs function.  I'm going to put this back to the way it was, and look into improving it as a follow up.